File: backend/app/schemas/scenario.py
from datetime import datetime
from pydantic import BaseModel


class ScenarioState(BaseModel):
    enabled: dict[str, bool]
    strength: dict[str, int]


class Scenario(BaseModel):
    id: str
    name: str
    updated_at: datetime
    state: ScenarioState

    class Config:
        from_attributes = True


class ScenarioCreate(BaseModel):
    name: str
    state: ScenarioState


class ScenarioUpdate(BaseModel):
    name: str | None = None
    state: ScenarioState | None = None

File: efce-ui/src/types/scenario.ts
// Copyright (c) 2026 Jeyapragash. All rights reserved.

export type ScenarioState = {
  enabled: Record<string, boolean>;
  strength: Record<string, number>;
};

export type ScenarioItem = {
  id: string;
  name: string;
  updatedAt: string;
  state: ScenarioState;
};

export type ScenarioCreate = {
  name: string;
  state: ScenarioState;
};

export type ScenarioUpdate = {
  name?: string;
  state?: ScenarioState;
};
